Error: The terraform-provider-azurerm_v4.30.0_x5 plugin crashed!

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.12.0

### AzureRM Provider Version

4.30.0

### Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
module "cluster" {
  source  = "Azure/aks/azurerm//v4"
  version = "10.0.1"

  cluster_name        = var.cluster_name
  resource_group_name = var.resource_group.name
  location            = var.resource_group.location
  prefix              = var.prefix

  log_analytics_workspace_enabled      = var.log_analytics_workspace_enabled
  log_analytics_workspace              = var.log_analytics_workspace
  log_analytics_solution               = var.log_analytics_solution
  role_based_access_control_enabled    = var.role_based_access_control_enabled
  rbac_aad                             = var.rbac_aad
  rbac_aad_admin_group_object_ids      = var.rbac_aad_admin_group_object_ids
  oidc_issuer_enabled                  = var.oidc_issuer_enabled
  enable_auto_scaling                  = var.enable_auto_scaling
  agents_max_count                     = var.node_pool_max_count
  agents_min_count                     = var.node_pool_min_count
  agents_max_pods                      = var.node_pool_max_pods
  agents_pool_name                     = var.node_pool_name
  agents_availability_zones            = var.availability_zones
  os_disk_size_gb                      = var.os_disk_size_gb
  agents_size                          = var.node_pool_instance_type
  agents_pool_drain_timeout_in_minutes = 30

  network_plugin             = var.aks_network_profile.network_plugin
  network_policy             = var.aks_network_profile.network_policy
  net_profile_dns_service_ip = var.aks_network_profile.dns_service_ip
  net_profile_service_cidr   = var.aks_network_profile.service_cidr

  vnet_subnet = {
    id = azurerm_subnet.subnet.id
  }

  kubernetes_version   = var.kubernetes_version
  orchestrator_version = var.orchestrator_version

  api_server_authorized_ip_ranges = var.public_access_cidrs

  client_id     = var.ARM_CLIENT_ID
  client_secret = var.ARM_CLIENT_SECRET

  automatic_channel_upgrade = var.automatic_channel_upgrade
  maintenance_window        = var.maintenance_window
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: Plugin did not respond
│ 
│   with module.cluster.module.cluster.azurerm_kubernetes_cluster.main,
│   on .terraform/modules/cluster.cluster/v4/main.tf line 13, in resource "azurerm_kubernetes_cluster" "main":
│   13: resource "azurerm_kubernetes_cluster" "main" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-azurerm_v4.30.0_x5 plugin:
panic: interface conversion: interface {} is nil, not map[string]interface {}
goroutine 83 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.expandKubernetesClusterAzureActiveDirectoryRoleBasedAccessControl({0xc003dddbc0?, 0x83559f6?, 0xc00008a363?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_resource.go:3584 +0x405
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.resourceKubernetesClusterUpdate(0xc003387c80, {0x71408c0, 0xc00229e488})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_resource.go:1946 +0xbe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x89fd438?, {0x89fd438?, 0xc003436660?}, 0xd?, {0x71408c0?, 0xc00229e488?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:860 +0x15f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0011e6000, {0x89fd438, 0xc003436660}, 0xc001f3ed00, 0xc003387b00, {0x71408c0, 0xc00229e488})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/resource.go:979 +0x84a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0006c8660, {0x89fd438?, 0xc0034365a0?}, 0xc00013a500)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.36.1/helper/schema/grpc_provider.go:1188 +0xd1a
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xc001f26580, {0x89fd438?, 0xc0034362d0?}, 0xc00013a500)
	github.com/hashicorp/terraform-plugin-mux@v0.17.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x18d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001f72960, {0x89fd438?, 0xc0033e9770?}, 0xc001a51490)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/tf5server/server.go:866 +0x3b9
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7ff7c60, 0xc001f72960}, {0x89fd438, 0xc0033e9770}, 0xc003386c80, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:611 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0009a6200, {0x89fd438, 0xc00392b470}, 0xc0000e03c0, 0xc000d2b680, 0xdbd4ff8, 0x0)
	google.golang.org/grpc@v1.69.4/server.go:1392 +0xfbe
google.golang.org/grpc.(*Server).handleStream(0xc0009a6200, {0x8a013c8, 0xc00134e1a0}, 0xc0000e03c0)
	google.golang.org/grpc@v1.69.4/server.go:1802 +0xb88
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.69.4/server.go:1030 +0x7f
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 12
	google.golang.org/grpc@v1.69.4/server.go:1041 +0x11d
Error: The terraform-provider-azurerm_v4.30.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

Terraform apply should run successfully for version 10.0.1 of the AKS module - https://registry.terraform.io/modules/Azure/aks/azurerm/10.0.1

I had to make some changes to my inputs to this module for the upgrade to v10. This involved:
- Removing variable "rbac_aad_managed"
- Converting "vnet_subnet_id" to an object "vnet_subnet"
- Adding required location variable

### Actual Behaviour

Terraform plan completes but when I run terraform apply I get the error mentioned in Debug Output

### Steps to Reproduce

terraform init
terraform plan
terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/containers/kubernetes_cluster_resource.go
+++ b/internal/services/containers/kubernetes_cluster_resource.go
@@ -2100,8 +2100,9 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 		azureADProfile := expandKubernetesClusterAzureActiveDirectoryRoleBasedAccessControl(azureADRaw)
 
 		props.AadProfile = azureADProfile
-		if props.AadProfile != nil && (props.AadProfile.Managed == nil || !*props.AadProfile.Managed) {
+		if props.AadProfile == nil || (props.AadProfile.Managed == nil || !*props.AadProfile.Managed) {
 			log.Printf("[DEBUG] Updating the RBAC AAD profile")
+			props.AadProfile = &managedclusters.ManagedClusterAADProfile{}
 			err = clusterClient.ResetAADProfileThenPoll(ctx, *id, *props.AadProfile)
 			if err != nil {
 				return fmt.Errorf("updating Managed Kubernetes Cluster AAD Profile for %s: %+v", *id, err)

